rskj-core/src/main/java/org/ethereum/vm/program/InternalTransaction.java

```diff
@@ -73,7 +79,11 @@ public String getNote() {
     }
 
     public byte[] getParentHash() {
-        return parentHash;
+        return parentHash.clone();
+    }
+
+    public byte[] getOriginHash() {
+        return originHash.clone();
```

Comment:
I don't see any use of this new information in current logic, only in test. Could you elaborate the rationale to add this info now? BTW, the concept of internal transaction is flawed in our implementation: there is no internal transaction as transacton, only message. The original implementors chose to inherits from Transaction, but really, in Ethereum world there is no internal transaction AS transaction, only as a message. That's the reason why Solidity has msg.sender, not tx.sender

Reply:
Thanks for taking a look at this Angel!
The originHash is not used in this PR. It will be used by the Bridge though.
The new fast-bridge logic will be consumed by contracts, so the executing transaction will be an InternalTransaction. The bridge needs the origin transaction hash to store it internally.
This information is used by the powpeg node to extract the release_requested log from said tx. (This is required by the peg-out process)

Reply:
Thanks for the description and information

I'm a bit afraid to use something (initial transaction hash) that the rest of Ethereum world does not use.

To me, the original sin was to inject a Transaction into Bridge init. In other circunstances (ie the add of TxIndex, now removed) we used EthereumJ ProgramInvoke.

But even if the tx hash is needed (ie in the bridge process (I don't see any other use in OTHER precompiled contracts), as in current code (tx hash is used in the process you mentioned), adding this feature to InternalTransaction maybe DON'T cover all the use case. IE, I could imagine a transaction calling a DeFi contract A, that internally makes MANY CALLS and at the end, produces MANY calls to bridge release process, using the SAME tx hash (and even the same repeated amount). That is one reason to not use transaction hash in Ethereum world: it does not identify practically anything, you need something more to refer to an event or work

I will contact you to discuess directly (in Spanish I hope :-))

Reply:
I'm with aj on this. It's hard to imagine reasons to use the origin tx hash and being something error-prone is a reason to not use it